### PR TITLE
add createmeta_84age implementation

### PR DESCRIPTION
**Pull request by dshvedchenko**
this is draft for implementation of createmeta required according to changes causing #1462

### CODE REVIEW

**Review by adehad — CHANGES_REQUESTED** [review]
Great start !!
Made some comments, it would also be great if we can get some tests in, let me know if you need help on that front.
I would have a look at the `tests/resources/` folder

File: jira/client.py [hunk 1]
```diff
@@ -1750,6 +1827,15 @@ def createmeta(
...
 
         """
+
+        if self._version >= (9, 0, 0):
+            raise NotImplementedError(
+                "Support for createmeta search in server" ">= 9.0.0 has been removed."
+            )
+
+        if self.is_84_and_later():
+            DeprecationWarning("The `createmeta` is deprecated, use createmeta_v2")
```
[the remark this answers is not in the document]
Comment (Ry-DS) [reply jira/client.py:1837]: @dshvedchenko sorry to hear you're affected 😔 hope all is well for you! If you give write access to your fork, I'm more than happy to finish the job :)

**Comment by dshvedchenko** [comment]
need suggestions on how to automate tests for this.
current ci server is still 8.17 so can not be used to test on 9.0.0 jira branch

File: jira/client.py [hunk 2]
```diff
@@ -1750,6 +1791,26 @@ def createmeta(
...
+        def deprecate_or_raise(msg: str):
+            if self._version >= (9, 0, 0):
+                raise ValueError(msg)
+            elif self._version >= (8, 4, 0):
+                DeprecationWarning(f"From Jira version 8.4.0: {msg}")
+
+        if not isinstance(projectKeys, str) or ("," in projectKeys):
+            deprecate_or_raise("projectKeys must be a str of a single project key")
+        elif self._version >= (9, 0, 0):
+            projectKey = projectKeys
```
[the remark this answers is not in the document]
Comment (dshvedchenko) [reply jira/client.py:1834]: I've fixed issueType id and Name and projected mandatoriness.

I also agree that reconstruct is not a best way, 

but this approach enables a lot of old client code to be in touch, and allow it to work with new server with only library version upgrade.

of course we need to provide dedicated method for 8.4.0+eon . btw it was my itial commit adding "prominent" :) createmeta_v2 method :)

[the remark this answers is not in the document]
Comment (dshvedchenko) [reply jira/client.py:1834]: remove dedicated method

**Comment by dshvedchenko** [comment]
can we have a progress on this or on #1527 PRs

**Comment by adehad** [comment]
Thank you again for all the effort on this PR @dshvedchenko, sorry that we went with the alternative PR this time. Hope you can contribute again in the future !

**Pull request closed without merging**